test/test_autograd.py

```diff
@@ -4064,6 +4064,60 @@ def basic_mul(x):
             with self.assertRaisesRegex(RuntimeError, err_msg):
                 gradcheck(bad_fn, (x, y), check_forward_ad=True, fast_mode=fast_mode)
 
+    def test_gradcheck_forward_ad_respects_requires_grad_but_runs_with_no_requires_grad(self):
+        # Currently requires_grad is used as a easy way for gradcheck to know
```

Comment:
Not sure what this test is checking? That the requires_grad flag is properly respected by the forward ad gradcheck?

Reply:
Hmm maybe I can improve the comments in this test here,
but yeah we want to make sure 1) requires_grad is properly respected by the various checks in gradcheck, and 2) that the actual test is run with requires_grad=False

for (1), we make sure the jvp count is what we expect, and for (2) we look at vjp to make sure it is 0 when check_backward_ad=False

Reply:
> (2) we look at vjp to make sure it is 0 when check_backward_ad=False

That only means that no backward() call happened. Which has always been True. It does not check that we don't run the forward with requires_grad=False right?
You could add a check in the function given to gradcheck to ensure that?